stdlib/Distributed/src/cluster.jl

```diff
@@ -572,13 +579,27 @@ end
 function create_worker(manager, wconfig)
     # only node 1 can add new nodes, since nobody else has the full list of address:port
     @assert LPROC.id == 1
-    timeout = worker_timeout()
 
     # initiate a connect. Does not wait for connection completion in case of TCP.
     w = Worker()
     local r_s, w_s
+
+
     try
-        (r_s, w_s) = connect(manager, w.id, wconfig)
+        rw_future = Future()
+        @async try
+            put!(rw_future, connect(manager, w.id, wconfig))
+        catch connect_error
+            put!(rw_future, connect_error)
```

Comment:
Move put! outside the try/catch, and rethrow on unexpected error types

Reply:
The name `connect_error` may be misleading. The intention was to handle any error here in the same manner.